node/node.go

```diff
@@ -776,23 +777,36 @@ func (n *Node) OpenDatabase(name string, cache, handles int, namespace string, r
 	return db, err
 }
 
-func (n *Node) OpenAndMergeDatabase(name string, cache, handles int, freezer, diff, namespace string, readonly, persistDiff, pruneAncientData bool) (ethdb.Database, error) {
-	chainDataHandles := handles
-	if persistDiff {
-		chainDataHandles = handles * chainDataHandlesPercentage / 100
+func (n *Node) OpenAndMergeDatabase(name string, namespace string, readonly bool, config *ethconfig.Config) (ethdb.Database, error) {
+	chainDataHandles := config.DatabaseHandles
+	if config.PersistDiff {
+		chainDataHandles = config.DatabaseHandles * chainDataHandlesPercentage / 100
 	}
-	chainDB, err := n.OpenDatabaseWithFreezer(name, cache, chainDataHandles, freezer, namespace, readonly, false, false, pruneAncientData)
+	var blockStore *leveldb.Database
+	var err error
+
+	if config.SeparateDB {
+		log.Info("SeparateDB is enabled, open block database, %s", config.DatabaseBlock)
+		blockStore, err = n.OpenBlockDatabase(name, config.DatabaseHandles-chainDataHandles, config.DatabaseBlock, namespace, readonly)
```

Comment:
config.DatabaseHandles-chainDataHandles euqals zero , it it no meaning.  
You can consider allocating 1/10 of the handlers to the block store